rxandroid/src/main/java/rx/android/lifecycle/LifecycleHelper.java

```diff
@@ -0,0 +1,92 @@
+package rx.android.lifecycle;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import rx.subjects.BehaviorSubject;
+
+/**
+ * This instance lives within the scope of an {@code Activity} in order to forward Lifecycle events
+ * to {@link rx.subjects.BehaviorSubject}
+ *
+ * Thread safe, since it's <i>Thread-confined</i> to the main Thread.
+ *
+ * @author Nabil Hachicha.
+ */
+class LifecycleHelper implements Application.ActivityLifecycleCallbacks {
+    private final Activity activityToMonitor;
+    private final BehaviorSubject<LifecycleEvent> lifecycleSubject;
+    /**
+     * Help track if {@link #onActivityCreated(Activity, Bundle)} was called
+     */
+    private boolean isCreateCalled = false;
+    /**
+     * Help track if {@link #onActivityStarted(Activity)} was called
+     */
+    private boolean isStarted = false;
+
+    public LifecycleHelper(Activity instance, BehaviorSubject<LifecycleEvent> lifecycle) {
+        activityToMonitor = instance;
+        lifecycleSubject = lifecycle;
+    }
+
+    @Override
+    public void onActivityCreated(Activity activity, Bundle bundle) {
+        // We may have independents instance of the same Activity in the activity stack
+        // this is why we use identity comparison
+        if (activity == activityToMonitor) {
+            lifecycleSubject.onNext(LifecycleEvent.CREATE);
+            isCreateCalled = true;
+        }
+    }
+
+    @Override
+    public void onActivityStarted(Activity activity) {
+        if (activity == activityToMonitor) {
+            if (!isCreateCalled) {//client bound onStart, sending missed event
```

Comment:
I question this behavior. If I bind in onStart I would expect to have missed onCreate. This is an application of lifecycle events which would be a hot observable. If you aren't around to observe the events, you don't get them.